### PR TITLE
Record: SP4096 + Int6 QAT + NorMuon (val_bpb=1.2012)

### PR DESCRIPTION
**Pull request by khasinski**
## Summary

- SP4096 tokenizer (26% better text compression than sp1024)
- Int6 STE QAT + zstd-22 compression (fits in 14.3MB)
- NorMuon optimizer + tuned learning rates

## Result

**val_bpb = 1.2012** on 8xH100 SXM, 600s training, 14.3MB artifact.

Beats baseline (1.2244) by 0.023 BPB.

## Key Metrics

| Metric | Value |
|--------|-------|
| val_bpb (post-quant) | 1.2012 |
| Artifact size | 14,342,773 bytes |
| Steps | 11,497 |
| Hardware | 8xH100 SXM 80GB |

## Test plan

- [x] Train sp4096 tokenizer and verify compression ratio
- [x] Local MLX smoke test
- [x] RTX 5090 10-min comparison (16 configs tested)
- [x] 8xH100 RunPod validation (600s)
- [x] Artifact under 16,000,000 bytes
- [x] train.log attached

### CODE REVIEW

**Comment by serdardoesml** [comment]
Would be nice to have the tokenizer and dataset somehow available to replicate. Also, this most likely wont scale up as the baseline compressed size is very close to the limit.
